src/coreclr/tools/Common/TypeSystem/Common/Instantiation.cs

```diff
@@ -113,5 +113,33 @@ public bool MoveNext()
                 return true;
             }
         }
+
+        public bool Equals(Instantiation other)
+        {
+            if (_genericParameters.Length != other._genericParameters.Length)
+                return false;
+
+            for (int i = 0; i < _genericParameters.Length; i++)
+            {
+                if (_genericParameters[i] != other._genericParameters[i])
+                    return false;
+            }
+            return true;
+        }
+        public override bool Equals(object o)
+        {
+            if (o is Instantiation inst)
+                return Equals(inst);
+            return false;
+        }
+        public override int GetHashCode()
+        {
+            int hashcode = 1;
+            foreach (var t in _genericParameters)
+            {
+                hashcode = BitOperations.RotateLeft(hashcode, 3) ^ t.GetHashCode();
+            }
+            return hashcode;
+        }
```

Comment:
We could also just do this:

```suggestion
        public override int GetHashCode() => ComputeGenericInstanceHashCode(1);
```

Reply:
Good ideas. It gets rid of the irritating BitOperations build failure.